torch/csrc/utils/tensor_new.cpp

```diff
@@ -535,6 +540,11 @@ Tensor legacy_tensor_new(c10::DispatchKey dispatch_key, at::ScalarType scalar_ty
     at::OptionalDeviceGuard device_guard(deviceOptional);
     return at::empty({0}, options(dispatch_key, scalar_type));
   } else if (r.idx == 1) {
+    at::ScalarType storage_scalar_type =
+        reinterpret_cast<THPDtype*>(
+            PyObject_GetAttrString(r.pyobject(0), "dtype"))
+            ->scalar_type;
+    TORCH_INTERNAL_ASSERT(storage_scalar_type == scalar_type);
```

Comment:
Should this really be an internal assert? This test will fail if I mismatch the storage type and the tensor type, right? So it should just be a `TORCH_CHECK`